java/src/jmri/jmrix/AbstractThrottle.java

```diff
@@ -1752,26 +1754,34 @@ protected void sendMomentaryFunctionGroup5() {
      * @param Mode the current speed step mode
      */
     @Override
-    public void setSpeedStepMode(int Mode) {
+    public void setSpeedStepMode(SpeedStepMode Mode) {
         log.debug("Speed Step Mode Change from:{} to Mode:{}",this.speedStepMode,Mode);
         if (speedStepMode != Mode) {
             notifyPropertyChangeListener("SpeedSteps", this.speedStepMode,
                     this.speedStepMode = Mode);
         }
-        if (Mode == DccThrottle.SpeedStepMode14) {
-            speedIncrement = SPEED_STEP_14_INCREMENT;
-        } else if (Mode == DccThrottle.SpeedStepMode27) {
-            speedIncrement = SPEED_STEP_27_INCREMENT;
-        } else if (Mode == DccThrottle.SpeedStepMode28) {
-            speedIncrement = SPEED_STEP_28_INCREMENT;
-        } else // default to 128 speed step mode
-        {
-            speedIncrement = SPEED_STEP_128_INCREMENT;
+        switch(Mode) {
+            case SpeedStepMode14:
+                speedIncrement = SPEED_STEP_14_INCREMENT;
+                break;
+            case SpeedStepMode27:
+                speedIncrement = SPEED_STEP_27_INCREMENT;
+                break;
+            case SpeedStepMode28:
+            case SpeedStepMode28Mot:
+                speedIncrement = SPEED_STEP_28_INCREMENT;
+                break;
+            case SpeedStepModeTMCC32:
+                speedIncrement = SPEED_STEP_32_INCREMENT;
+                break;
+            case SpeedStepMode128:
```

Comment:
You should add a default after this case to ensure the switch remains exhaustive even if another speed step is added later. It can be combined with it the 128 step case.

Reply:
The Travis CI tests are indicating that a default clause is required.

Reply:
Yeah; I'm still working my way through the test failures from Travis.
I think I'm just going to make the number of steps and increment part of the SpeedStepMode enum and blow away the code that needs to infer the increment or number of steps.